koku/masu/util/common.py

```diff
@@ -534,11 +534,22 @@ def fetch_optional_columns(local_file, current_columns, fetch_columns, tracing_i
     """Add optional columns to columns list if they exists in files"""
     for fetch_column in fetch_columns:
         try:
-            data_frame = pd.read_csv(local_file, usecols=lambda col: col.lower().startswith(fetch_column.lower()))
-            data_frame = data_frame.dropna(axis=1, how="all")
-            fetch_cols = data_frame.columns
-            for col in fetch_cols:
-                current_columns.add(col)
+            data_frame = pd.read_csv(
+                local_file, usecols=lambda col: col.lower().startswith(fetch_column.lower()), nrows=0
+            )
+            header_chunks = chunk_columns(data_frame.columns, settings.PANDAS_COLUMN_BATCH_SIZE)
```

Comment:
Isn't `fetch_column` a single column? If that is the case, do we need to do the column batching?

Reply:
So this is doing starts with `reousrceTags` and for AWS that can match hundreds of columns depending on tag counts for a customer. Since aws has a column per tag. I ran one of our large customers and not batching we used around 5G, but with batching we stayed under 2G. :smile: